src/renderer.rs

```diff
@@ -348,10 +348,31 @@ impl Renderer {
         self.index_buffer = index_buffer;
     }
 
-    fn copy_mesh(&self, mesh: Mesh, vertex_start: DeviceSize, index_start: DeviceSize) {
+    fn copy_mesh(&mut self, mesh: Mesh, vertex_start: DeviceSize, index_start: DeviceSize) {
         // Copy vertices to buffer
         let v_slice = &mesh.vertices;
-        let mut vertex_content = self.vertex_buffer.write().unwrap();
+        // Attempt to write to the old vertex buffer. It might be in use, in which case we simply create a new one.
+        // TODO: consider a double buffering solution perhaps.
+        let vc = self.vertex_buffer.write();
+        // Now we want to allocate new buffers if they are in use.
+        // We can't use match here due to holding a reference to self.vertex_buffer.
+        let mut vertex_content = if vc.is_err() {
```

Comment:
Agreed, this is an ugly hack to recover form GpuLocked error. In my oppinion, it would be better with a double buffering solution, see separate comment for my reasoning.